pkg/platform/openstack/networking.go

```diff
@@ -183,7 +183,15 @@ func findOpenStackRouter(client *gophercloud.ServiceClient, name, tag string) (r
 	if len(routerList) == 1 {
 		return routerList[0], nil
 	} else if len(routerList) == 0 {
-		return empty, errors.New("router not found")
+		router, err := routers.Create(client, routers.CreateOpts{Name: name}).Extract()
```

Comment:
we also need to connect the router to the VMs/Nodes network, right?

Reply:
True!